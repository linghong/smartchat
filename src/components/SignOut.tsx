import React from 'react';
import { useRouter } from 'next/router';
import { LogOut } from 'lucide-react';

const SignOut: React.FC = () => {
  const router = useRouter();

  const handleSignOut = () => {
    localStorage.removeItem('token');
    router.push('/login');
  };

  return (
    <button
      onClick={handleSignOut}
      className="flex items-center justify-center p-1 text-white text-sm rounded-full hover:bg-stone-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-stone-500"
      aria-label="Sign out"
    >
      <LogOut size={16} aria-label="logout" />
      <span className="ml-1">Log out</span>
    </button>
  );
};

export default SignOut;
