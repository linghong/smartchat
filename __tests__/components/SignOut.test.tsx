import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SignOut from '@/src/components/SignOut';
// Mock the next/router module
jest.mock('next/router', () => ({
  useRouter: jest.fn()
}));
// Import the mocked useRouter
import { useRouter } from 'next/router';
// Mock the window.localStorage
const localStorageMock = (function () {
  let store: { [key: string]: string } = {};
  return {
    getItem: function (key: string) {
      return store[key] || null;
    },
    setItem: function (key: string, value: string) {
      store[key] = value.toString();
    },
    removeItem: function (key: string) {
      delete store[key];
    },
    clear: function () {
      store = {};
    }
  };
})();
Object.defineProperty(window, 'localStorage', {
  value: localStorageMock
});
describe('SignOut Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<SignOut />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    render(<SignOut />);
    expect(screen.getByText('Log out')).toBeInTheDocument();
  });

  it('has the correct icon', () => {
    render(<SignOut />);
    const icon = document.querySelector('svg');
    expect(icon).toBeInTheDocument();
  });

  it('calls handleSignOut when clicked', () => {
    const pushMock = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      push: pushMock
    });
    render(<SignOut />);
    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(localStorage.getItem('token')).toBeNull();
    expect(pushMock).toHaveBeenCalledWith('/login');
  });
});
