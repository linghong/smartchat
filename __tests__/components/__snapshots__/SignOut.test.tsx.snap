// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOut Component renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="Sign out"
    class="flex items-center justify-center p-1 text-white text-sm rounded-full hover:bg-stone-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-stone-500"
  >
    <svg
      aria-label="logout"
      class="lucide lucide-log-out"
      fill="none"
      height="16"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
      />
      <polyline
        points="16 17 21 12 16 7"
      />
      <line
        x1="21"
        x2="9"
        y1="12"
        y2="12"
      />
    </svg>
    <span
      class="ml-1"
    >
      Log out
    </span>
  </button>
</DocumentFragment>
`;
